apps/backend/src/controllers/api/userRouter.ts

```diff
@@ -14,35 +16,77 @@ userRouter.post(
   "/",
   async (req: Request, res: Response, next: NextFunction) => {
     try {
-      const { email, username, password } = req.body;
-      if (!email.length) {
-        res.status(400).send({ error: "Your email cannot be blank" });
+      const { email, username, password, oauth, role } = req.body;
+      const cleanedEmail = email?.trim();
+      const cleanedUsername = username?.trim();
+      const cleanedPassword = password?.trim(); 
+      const cleanedRole = role?.trim();
+
+      if (!cleanedEmail?.length) {
+        res.status(400).send({ error: "Your email cannot be blank." });
+        return;
+      }
+
+      if (!cleanedUsername?.length) {
+        res.status(400).send({ error: "Your username cannot be blank." });
+        return;
+      }
+
+      if (oauth === undefined && !cleanedPassword?.length) {
+        res.status(400).send({ error: "Your password cannot be blank." });
         return;
       }
 
-      if (!password.length) {
-        res.status(400).send({ error: "Your password cannot be blank" });
+      if (!cleanedRole?.length) {
+        console.log(cleanedRole);
+        res.status(400).send({ error: "Your role cannot be blank." });
         return;
       }
 
-      if (!username.length) {
-        res.status(400).send({ error: "Your username cannot be blank" });
+      if (!Object.values(Role).includes(cleanedRole as Role)) {
+        console.log(cleanedRole as Role);
+        res.status(400).send({ error: `Invalid role: ${cleanedRole}` });
         return;
       }
 
-      const hasSameEmailUser = await findOneUser({ email });
+      const hasSameEmailUser = await findOneUser({ email: cleanedEmail });
       if (hasSameEmailUser) {
-        res.status(400).send({ error: "That email is already in use" });
+        res.status(400).send({ error: "That email is already in use." });
         return;
       }
 
-      const hasSameUsernameUser = await findOneUser({ username });
+      const hasSameUsernameUser = await findOneUser({ username: cleanedUsername });
       if (hasSameUsernameUser) {
-        res.status(400).send({ error: "That username is already in use" });
+        res.status(400).send({ error: "That username is already in use." });
         return;
       }
 
-      const newUser = await createUser(req.body);
+      const cleanedOauth: OAuth[] = [];
+      const invalidOauth: string[] = [];
+      if (oauth !== undefined) {
+        for (const auth of oauth) {
+          if (!Object.values(OAuthType).includes(auth as OAuthType)) {
+            invalidOauth.push(auth);
+            continue;
+          }
+          cleanedOauth.push(auth as OAuthType);
+        }
+      }
+
+      if (invalidOauth.length !== 0) {
+        console.warn(`The following OAuths are invalid and are ignored: ${invalidOauth}`);

```

Comment:
```suggestion
        logger.warn(`The following OAuths are invalid and are ignored: ${invalidOauth}`);

```
Change this to the logger used in the backend service